build/tasks/typescript.js

```diff
@@ -27,118 +27,109 @@ var runSequence = require('run-sequence');
-    'src/clarity/**/*.html',
-    '!src/clarity/**/demo/**'
+    'src/clarity-angular/**/*.ts',
+    '!src/clarity-angular/**/*.spec.ts',
+    '!src/clarity-angular/**/*.mock.ts'
 ];
 
-gulp.task('typescript:clarity', ['triage:typings', 'triage:clarity', 'triage:icons'], function () {
-    return typescriptCompile(['tmp/clarity-angular/**/*.ts'], { // triage will generate these
+gulp.task('typescript:clarity', function () {
+    return typescriptCompile(claritySources, {
         inlineTemplates: true
-    });
-});
-
-/**
- * Demo components
- */
-var demosSources = [
-    'src/clarity/**/demo/**/*.ts',
-    'src/clarity/**/demo/**/*.html'
-];
-
-gulp.task('typescript:demos', ['triage:typings', 'triage:clarity', 'triage:demos', 'triage:icons'], function () {
-    return typescriptCompile(['tmp/clarity-demos/**/*.ts'], { // triage will generate these
-        inlineTemplates: true
-    });
+    })
+    .pipe(gulp.dest("dist"));
 });
 
 /**
  * Tests
  */
-var testsSources = ['src/clarity/**/*.spec.ts', 'src/clarity/**/*.mock.ts', 'src/testing/**/*.ts'];
+var testsSources = ['src/clarity-angular/**/*.spec.ts', 'src/clarity-angular/**/*.mock.ts'];
 
-gulp.task('typescript:tests', ['triage:typings', 'triage:clarity', 'triage:tests'], function () {
-    return typescriptCompile(['tmp/tests/**/*.ts'], { // triage will generate these
+gulp.task('typescript:tests', function () {
+    return typescriptCompile(testsSources, {
         inlineTemplates: false,
         internal: true
-    });
+    })
+    .pipe(absoluteRequires({
+        pattern: /\.\.?\/.*(mock|spec)/,
+        rename: {
+            "clarity-angular": "tests"
+        }
+    }))
+    // The requires remaining are all actual clarity classes
+    .pipe(absoluteRequires({}))
+    .pipe(renameFolder({"clarity-angular": "tests"}))
+    .pipe(gulp.dest("dist"));
 });
 
 
 /**
  * Sample application
  */
-var appSources = ['src/app/**/*.ts', '!src/app/routes.ts'];
+var appSources = ['src/app/**/*.ts'];
 
-gulp.task('typescript:app', ['triage:typings', 'triage:clarity', 'triage:demos', 'triage:icons', 'triage:app'], function () {
-    return typescriptCompile(['tmp/app/**/*.ts'], { // triage will generate these
+gulp.task('typescript:app', function () {
+    return typescriptCompile(appSources, {
         inlineTemplates: false,
         internal: true
-    });
+    })
+    .pipe(absoluteRequires({
+        parentOnly: true
+    }))
+    .pipe(gulp.dest("dist"));
 });
 
 /**
  * Watches for changes in ts files (or files that will be inlined in typescript)
  * to retrigger typescript compilation.
  */
+var clarityHtmlFiles = ["src/clarity-angular/**/*.html"];
 gulp.task('typescript:clarity:watch', function () {
-    gulp.watch(claritySources, function () {
-        return runSequence('tslint:watch', 'typescript:clarity');
+    gulp.watch(claritySources.concat(clarityHtmlFiles), function () {
+        return runSequence('tslint:clarity:watch', 'typescript:clarity');
```

Comment:
We could have a separate watcher on HTML that doesn't re-run tslint, if it becomes too slow at some point.

PS: `"tslint:clarity:watch"` is misleading, it made me pause and go look again because I thought our watcher was starting a new watcher every time. Maybe `"tslint:clarity:no-errors"` or something similar would be clearer.